File: Gombang/[LeetCode75] - [Stack]/2390_Removing_Stars_From_a_String.cpp
class Solution {
public:
    string removeStars(string s) {
        stack<char> stack;

        for (char ch : s)
        {
            if (ch == '*')
                stack.pop();
            else
                stack.push(ch);
        }

        string str;
        while (stack.empty() == false)
        {
            str += stack.top();
            stack.pop();
        }

        reverse(str.begin(), str.end());

        return str;
    }
};

File: Lucy/2025/W03/2390_Removing_Stars_From_a_String.ts
function removeStars(s: string): string {
  const STAR = "*";

  const stack: string[] = [];
  for (const ch of s) {
    if (ch === STAR) {
      stack.pop();
    } else {
      stack.push(ch);
    }
  }

  return stack.join("");
}

removeStars("leet**cod*e"); // "lecoe"
removeStars("erase*****"); // ""
